### ISSUE TITLE
tf.nn.batch_nor malization and tf.keras.layers.BatchNormalization outputs has a diff (10-3 range)

### ISSUE THREAD

**Issue by yogeesh-alphaics** [keras-team/keras#15571]
System information

Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
OS Platform and Distribution (e.g., Linux Ubuntu 16.04): 20.04
I have implemented a classification model , which has Batchnormalization layers into it. While training the model , i used ```tf.keras.layers.BatchNormalization``` layer for training. Then manually extracted the bn weights (gamma, beta, mean , var) from checkpoint file and called the bn using ```tf.nn.batch_normalization``` layer using above statistics for inference. I observed a drop in testing accuracy (~75%) as compared to the scenario where i load the weights in the model using keras itself model.load_weights(chkpnt_file) (~92%).
Though if i replace the ```tf.nn.batch_normalization``` call to keras bn layer call by calling set_weights function , both the cases gived 0.0 diff, which apprently led me to believe that there is diff in both the calls while the input and bn_weights are exactly same. epsilon is also same.

Standalone code to reproduce the issue
Please refer to this colab notebook(https://colab.research.google.com/drive/1nysYyCR5Ay_jkPlh1WD9A1Mf01eVvhMu? usp=sharing) for a small example. Here the error is coming to be a margin of 10^-3 which theoretically shouldn't be there.

**Comment by sushreebarsa** [comment]
@jvishnuvardhan Was able to reproduce the issue on colab using TF v2.6.0  & tf-nightly ,please find the [gist](https://colab.research.google.com/gist/sushreebarsa/b8d723c9212e806964142da462459dc2/untitled14.ipynb#scrollTo=45dbaOsur9cS) for reference.Thanks!

**Comment by yogeesh-agarwal** [comment]
is there any update on this ? @sushreebarsa  , @jvishnuvardhan

**Comment by funkyyyyyy** [comment]
facing a related issue where

on CPU

given the same weights and variable `epsilon` value
- gamma
- beta
- moving mean
- moving var
- epsilon

`tf.keras.layers.BatchNormalization` output is different from batch norm formula calculation using `numpy` by `e^-2` range

for reference, numpy code is attached below

```
out_formula = batch-moving_mean
out_formula = out_formula / np.sqrt(moving_var + epsilon)
out_formula = gamma * out_formula
out_formula = out_formula + beta
```

where `batch` is input to `batch_norm` layer

**Comment by funkyyyyyy** [comment]
If its any help, I dug deeper into the Keras code and found the source of discrepancy

if the input shape is (N, C) or (N, T, C), then `tf.nn.batch_normalization` is used within `tf.keras.layers.BatchNormalization` to calculate the batch normalization. The output value here is in line with numpy outputs. 

However, if the input shape is longer such as (N, H, W, C) or (N, T, H, W, C), then `tf.compat.v1.nn.fused_batch_norm` is used and the large discrepancy in output occurs

**Comment by jvishnuvardhan** [comment]
Duplicate of # https://github.com/keras-team/keras/issues/15592

Please check the above issue and follow the code owner response. thanks!
